### PR TITLE
Please... we need a rustup!

### PR DESCRIPTION
**Pull request by Wafflespeanut**
Something's wrong with `let_and_return` and `let_unit_value`. They seem to disturb some of the `for` loops in tests. o_O

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
Whoa! Look at that! 83 errors??!!! O_O

**Comment by Manishearth** [comment]
We need to be able to check the source of an expansion for our macro-checking to work.

Side note: Our tests should have caught that too. We need to improve those.

File: tests/compile-fail/for_loop.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ impl Unrelated {
 
 #[deny(needless_range_loop, explicit_iter_loop, iter_next_loop, reverse_range_loop, explicit_counter_loop)]
 #[deny(unused_collect)]
-#[allow(linkedlist,shadow_unrelated,unnecessary_mut_passed)]
+#[allow(let_and_return, let_unit_value, linkedlist, shadow_unrelated, unnecessary_mut_passed)]
```
Comment (Manishearth) [inline tests/compile-fail/for_loop.rs]: This file should not be triggering that lint at all.

File: src/utils.rs [hunk 2]
```diff
@@ -18,12 +18,7 @@ pub const LL_PATH:     [&'static str; 3] = ["collections", "linked_list", "Linke
 /// returns true this expn_info was expanded by any macro
 pub fn in_macro(cx: &LateContext, span: Span) -> bool {
     cx.sess().codemap().with_expn_info(span.expn_id,
-            |info| info.map_or(false, |i| {
-        match i.callee.format {
-            ExpnFormat::CompilerExpansion(..) => false,
-            _ => true,
-        }
-    }))
+            |info| info.map_or(false, |_| true))
```
Comment (fhartwig) [inline src/utils.rs]: This can be written as "info.is_some()", right?

**Comment by Manishearth** [comment]
I'm working on a fix, it's a bit more involved that usual.

**Pull request closed without merging**